### ISSUE TITLE
StorageClass issue

### ISSUE THREAD

**Issue by aboeglin** [komljen/helm-charts#11]
According to comments in this post : https://akomljen.com/get-kubernetes-logs-with-efk-stack-in-5-minutes/, it seems a lot of people struggle to get it running.

```
k get pods -n logging                                      
NAME                                                 READY   STATUS    RESTARTS   AGE
efk-kibana-fc4bbccb7-ppc9x                           0/1     Running   0          14m
es-client-efk-cluster-857fd4c567-ngfwn               0/1     Running   3          13m
es-data-efk-cluster-default-0                        0/1     Pending   0          13m
es-master-efk-cluster-default-0                      0/1     Running   0          13m
es-operator-elasticsearch-operator-fbbd9556c-wv8jn   1/1     Running   1          19m
fluent-bit-b65vt                                     1/1     Running   0          14m
fluent-bit-rx2gk                                     1/1     Running   0          14m
```

I'm currently in such a state as well. I installed the chart through terraform helm provider. Installation went fine, but it doesn't run as the volumes couldn't be created. Would it be possible to expose a storage class or an option in the values file ? And document that so it would work with other IaaS than aws ?

**Comment by komljen** [comment]
Makes sense, I will work on that probably over the weekend. Thanks for opening the issue!

**Comment by LenXdata** [comment]
Hello, today I installed efk-stack from your tutorial "Get Kubernetes Logs with EFK Stack in 5 Minutes". But not everything started. Persistent Volume Claims are not created. Because classStorage is most likely missing. How to fix? I would be grateful for the answer.

![image](https://user-images.githubusercontent.com/46480200/52362605-40358180-2a41-11e9-8c99-bb42a7d04a50.png)

![image](https://user-images.githubusercontent.com/46480200/52362647-54797e80-2a41-11e9-9676-af965992e84f.png)

**Comment by JWebDev** [comment]
Hello, today I installed efk-stack from your tutorial "Get Kubernetes Logs with EFK Stack in 5 Minutes". But not everything started. Persistent Volume Claims are not created. Because classStorage is most likely missing. How to fix? I would be grateful for the answer.

![image](https://user-images.githubusercontent.com/1653372/52362954-fd27de00-2a41-11e9-8571-944e6aa65335.png)

![image](https://user-images.githubusercontent.com/1653372/52362982-0add6380-2a42-11e9-9b55-818ee1faebb4.png)

![image](https://user-images.githubusercontent.com/1653372/52363003-16308f00-2a42-11e9-9512-ac3186fa9b68.png)

**Comment by komljen** [comment]
Yeah, a storage class is probably missing, but it is required. In which env did you run the k8s and do you have access to some distributed storage?

**Comment by JWebDev** [comment]
This is most likely my problem. My cluster works on bare metal.
I have a master and three nodes.

Now I try to create nfs-server. But so far not very successful.

Tell me please, is it possible to somehow install efk-stack without network file storage?
Does the default installation take default storageClass?
On the cluster must be present dynamic provisioning?
Is it possible to somehow configure this all to install efk-stack by specifying the folder for storing elasticsearch and that's it?

Thanks for answers.

**Comment by JWebDev** [comment]
Everything works. Thanks. I had a problem with the NFS server.
Tell me please, is this version of kibana with x-packs?
How to install them?

**Comment by komljen** [comment]
This is kibana-oss image without x-pack. But, you could set an image with x-pack here https://github.com/komljen/helm-charts/blob/master/efk/values.yaml#L6 to `docker.elastic.co/kibana/kibana`.
